jupyterhub/templates/proxy/deployment.yaml

```diff
@@ -104,11 +105,13 @@ spec:
           resources:
             {{- .Values.proxy.chp.resources | toYaml | trimSuffix "\n" | nindent 12 }}
           env:
+            # Restarting the proxy pod causes a temporary outage,
+            # so we should do it as few times as possible. Including
+            # the auth token here directly means the pod only restarts
+            # if the auth token changes, rather than when the hub-secret
+            # changes, since the latter happens all the time.
             - name: CONFIGPROXY_AUTH_TOKEN
-              valueFrom:
-                secretKeyRef:
-                  name: {{ include "jupyterhub.hub-secret.fullname" . }}
-                  key: JupyterHub.proxy_auth_token
+              value: {{ include "jupyterhub.config.JupyterHub.proxy_auth_token" . | quote }}
```

Comment:
👍 for reasoning, requesting review from @consideRatio to check that the value will still be right in all cases here, or if more logic is needed.

Or if we should duplicate this value to the proxy-secret as well so it's not directly in the pod manifest.

Reply:
Especially with the lookup logic for autogenerating them - I don't know if it'll get generated twice?

Reply:
Hmmm, it will be a transient issue, causing one unwanted restart.

It will generate the same value as long as its not doing the first installation. So when the upgrade following a seeded installation happen it will align with what is within the hub k8s secret but then remain stable.